sunpy/net/jsoc/tests/test_jsoc.py

```diff
@@ -355,3 +355,15 @@ def test_check_request_keywords(client):
         a.Time('2020/1/1T1:00:36', '2020/1/1T01:00:38'),
         a.jsoc.Series('aia.lev1_euv_12s'), a.jsoc.Keyword("QUALITY") < 2, a.jsoc.Keyword("EXPTIME") > 2)
     assert len(responses) == 7
+
+
+@pytest.mark.remote_data
+def test_empty_response_fetch(client):
+    # Check that results from downloading an empty response are empty
+    response = client.search(
+        a.Time('1990-12-12T00:00:00', '1990-12-12T0:01:00'),
+        a.jsoc.Series.hmi_b_720s
+    )
+    assert len(response) == 0
```

Comment:
Can we not mock a response here to test this behaviour in an offline test?

Something like this should work:
```suggestion
    response = JSOCResponse()
```

Reply:
We could, but I think it's safer here to test the code in the way a user would interact with it.

Reply:
I don't think anything that actually involves interacting with JSOC servers is a good test :stuck_out_tongue: 

If we can avoid having to make a call to JSOC then the test is going to be a lot more reliable and less likely to be ignored as a "transient failure".

If you want to have both then ok, but I would rather have one which is offline as well.

Reply:
👍  - I've done as suggested, but left the search in a comment so it's easy to enable an end-to-end test in the future (e.g. if the `vcrpy` stuff ever gets sorted).